### ISSUE TITLE
opm migrate doesn't capture maxOpenShiftVersion property

### ISSUE THREAD

**Issue by jdockter** [operator-framework/operator-registry#921]
Using opm 1.20 I ran the following:

```
mkdir ibm-catalog
opm migrate icr.io/cpopen/ibm-operator-catalog:latest ibm-catalog
cd ibm-catalog/ibm-watson-speech-operator
cat catalog.json | jq 'select(.schema == "olm.bundle" and .image == "icr.io/cpopen/ibm-watson-speech-operator-bundle@sha256:79ed1c96ac00c3dcd386340222ccc6b61399ca4f67a375289b0c2b135b2a69cc")' | grep maxOpenShiftVersion
```

I do not find any elements representing `olm.maxOpenShiftVersion` under the properties array.  However when I render a bundle directly the value is captured.

```
opm render icr.io/cpopen/ibm-watson-speech-operator-bundle@sha256:79ed1c96ac00c3dcd386340222ccc6b61399ca4f67a375289b0c2b135b2a69cc > ibm-watson-speech-operator.v4.0.5.json
cat ibm-watson-speech-operator.v4.0.5.json | grep maxOpenShiftVersion -A2 -B1

 {
            "type": "olm.maxOpenShiftVersion",
            "value": "4.8"
        },
```

**Comment by jdockter** [comment]
@joelanford @cdjohnson

**Comment by joelanford** [comment]
@jdockter I don't think I'm seeing those properties in the `/database/index.db` file's `properties` or `dependencies` table (which is where I'd expect them to be sourced from), and I don't see them in the GRPC ListBundles response (which is effectively what `migrate` uses) to convert from sqlite to FBC.

So I don't think there's a bug with `migrate` here. However, maybe there's a bug with `opm index|registry add`?

**Comment by jdockter** [comment]
@joelanford ... I'm using opm 1.15.4 which doesn't seem to honor adding that property to the database from a bundle.  In testing different versions it looks like I need opm 1.18.1 which I think came out with 4.9 correct?  In 1.18.1 I see a database migration.  Any reason to think this is going to be an issue to run a catalog built with 1.18.1 on OCP 4.6?

**Comment by joelanford** [comment]
I'd have to defer to @kevinrizza and @dinhxuanvu on that question. Our general advice is to keep the `opm` and OLM versions in sync with each other, so I can't say for sure that that would work.

Separately, I personally lost the thread on what (if anything) OLM does on-cluster with the `olm.maxOpenShiftVersion` property. I know that some catalogs have separate tags per OpenShift version and that property is used to include/exclude bundles at catalog build time. In that case a cluster using those catalogs will ostensibly only see bundle versions that are known in advance to be compatible with it. Perhaps @perdasilva has some details about that.

**Comment by jdockter** [comment]
Found a thread [here ](https://kubernetes.slack.com/archives/C0181L6JYQ2/p1634155894183400) and [here](https://kubernetes.slack.com/archives/C0181L6JYQ2/p1634142026178500).  I guess it was as early as 1.17.1, maybe 1.17.4 for properties.yaml

**Comment by jdockter** [comment]
This can be closed as using upstream opm 1.17.5 has resolved the issue.
